dev/README.md

```diff
@@ -0,0 +1,39 @@
+## Development Environment
```

Comment:
We already have some development documentation:
https://github.com/astronomer/astro-sdk/tree/main/docs

It may be worth extending it

Reply:
moved the documentation in [docs](https://github.com/astronomer/astro-sdk/tree/main/docs) folder